### PR TITLE
Fix warnings & refactoring

### PR DESCRIPTION
**Pull request by Tykvik**
Fix warnings.
Code refactoring.

### CODE REVIEW

File: src/Format/Msg/File.cpp [hunk 1]
```diff
@@ -123,28 +125,25 @@ void File::_initialize()
...
-    for (auto message : _messages)
-    {
-        if (message->number() == number)
-        {
-            return message;
-        }
-    }
+    auto message_iterator = _messages_map.find(number);
+    if (message_iterator != _messages_map.end())
+        return message_iterator->second;
```
Comment (phobos2077) [inline src/Format/Msg/File.cpp]: Missing brackets.

Comment (Tykvik) [inline src/Format/Msg/File.cpp]: fixed

File: src/Format/Map/File.cpp [hunk 2]
```diff
@@ -212,7 +212,7 @@ void File::_initialize()
 
             if (object->inventorySize() > 0)
             {
-                for (unsigned int i = 0; i != object->inventorySize(); ++i)
+                for (unsigned int inv_counter = 0; inv_counter != object->inventorySize(); ++inv_counter)
```
Comment (phobos2077) [inline src/Format/Map/File.cpp]: Use camelCase for variables.

File: src/Format/Msg/File.cpp [hunk 3]
```diff
@@ -123,27 +125,26 @@ void File::_initialize()
...
 {
-    return &_messages;
+    return &_messages_vector;
 }
 
-Message* File::message(unsigned int number)
+Message* File::message(unsigned long number)
 {
-    for (auto message : _messages)
+    auto message_iterator = _messages_map.find(number);
```
Comment (phobos2077) [inline src/Format/Msg/File.cpp]: Variable naming.

File: src/Format/Msg/File.h [hunk 4]
```diff
@@ -51,10 +51,11 @@ class File : public Dat::Item
     ~File();
 
     std::vector<Message*>* messages();
-    Message* message(unsigned int number);
+    Message* message(unsigned long number);
 
 protected:
-    std::vector<Message*> _messages;
+    std::map<unsigned long, Message*> _messages_map;
+    std::vector<Message*>             _messages_vector;
```
Comment (phobos2077) [inline src/Format/Msg/File.h]: Looking at the class declaration, it's not clear why you need 2 containers for messages. Add comments for each new member. And fix style.

File: src/Format/Msg/Message.cpp [hunk 5]
```diff
@@ -44,12 +44,12 @@ Message::~Message()
 {
 }
 
-void Message::setNumber(unsigned int number)
+void Message::setNumber(unsigned long number)
```
Comment (phobos2077) [inline src/Format/Msg/Message.cpp]: Any specific reason to use `long` instead of `int`?

**Comment by phobos2077** [comment]
@Tykvik, All build warnings are fixed now in master. What was your original goal with this PR?

**Comment by Zervox** [comment]
void MvePlayer::_setPalette(uint8_t* data)
should use 
for (uint16_t i = start; i <= count; i++)

instead of 
for (uint8_t i = start; i <= count; i++)

**Comment by phobos2077** [comment]
@Zervox : you can comment on line of code directly, instead of copying code.

**Pull request closed without merging**